coap-gateway/service/resourceDirectory.go

```diff
@@ -65,7 +72,7 @@ func resourceDirectoryPublishHandler(req *mux.Message, client *Client) {
 		return
 	}
 
-	var w wkRd
+	w := makeWkRd()
```

Comment:
Why to define a new func which is used just once and not directly assign
```
w := &wkRd{
		TimeToLive:       -1,
		TimeToLiveLegacy: -1,
	}
```

Reply:
`TimeToLive` and `TimeToLiveLegacy` should always be initialized to -1, I have no way of enforcing it since Go doesn't do constructors, so this pseudo-constructor should always be used when wkRd is being initialized. Though this is not ideal since a dev can not notice the pseudo-constructor and call `&wkRd{...}` directly and not set the `-1` values. We could change `TimeToLive` and `TimeToLiveLegacy` to pointers or translate 0 to some other value that would signify unlimited TTL if this seems too fragile.